x-pack/plugins/security_solution/public/common/components/grouping/groups_selector/custom_field_panel.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EuiComboBoxOptionOption } from '@elastic/eui';
+import { EuiButton, EuiComboBox, EuiForm, EuiFormRow } from '@elastic/eui';
+import type { FieldSpec } from '@kbn/data-views-plugin/common';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+
+export interface GroupByOptions {
+  text: string;
+  field: string;
+}
+
+interface Props {
+  onSubmit: (field: string) => void;
+  fields: FieldSpec[];
+  currentOptions: GroupByOptions[];
+}
+
+interface SelectedOption {
+  label: string;
+}
+
+const initialState = {
+  selectedOptions: [] as SelectedOption[],
+};
+
+type State = Readonly<typeof initialState>;
+
+export class CustomFieldPanel extends React.PureComponent<Props, State> {
+  public static displayName = 'CustomFieldPanel';
+  public readonly state: State = initialState;
+  public render() {
+    const { fields, currentOptions } = this.props;
+    const options = fields
+      .filter(
+        (f) =>
+          f.aggregatable &&
+          f.type === 'string' &&
+          !(currentOptions && currentOptions.some((o) => o.field === f.name))
+      )
+      .map((f) => ({ label: f.name }));
```

Comment:
```suggestion
      .map<EuiComboBoxOptionOption>((f) => ({ label: f.name }));
```

And you can get rid of the cast on line `75` below.